source/deployment/permissions-backend.rst

```diff
@@ -0,0 +1,104 @@
+
+- A channel context has a parent team context, whose parent is the system context. For example, the "Developers Hangout" channel is the channel context, with parent team context "Contributors Team", with parent system context.
+- A team context has a parent system context and child channel contexts. For example, the "Contributors Team" is the team context, with parent system context, and with children channel contexts such as "Developers Hangout", "Reception" and "Marketing".
+
+When determining whether a user is allowed to carry out a given action in a given context, the union of the permissions of all roles that user has been assigned in the current context and its parent contexts is calculated. This enables permissions to cascade down the scope hierarchy. For example, if a users is granted the ``manage_public_channel_properties`` permission in a role in the system context, then the user has permissions to manage public channel properties in all channels, in all teams, of which they are a member.
+
+Schemes
+~~~~~~~~~
+
+Schemes describe the default roles applied to users in a context, and all child contexts. Schemes are either defined specifically for a context, or if they are not specified, the relevant parts of the parent context’s scheme are applied, ultimately climbing the hierarchy to the System Scheme, which serves the purpose of providing the system-wide defaults. For example, if Team A does not have a team-scoped scheme defined, the System Scheme will provide the defaults for all contexts in Team A.
+
+Additionally, the lowest-scoped scheme always takes precedence in the context. For example, if Team B has a team-scoped scheme, that scheme takes precedence over the System Scheme defaults for all contexts in Team B. 
+
+Data Structure
+----------------
+
+Permissions
+~~~~~~~~~~~~
+
+Permissions in Mattermost are a property of the server code base and are not created or modified dynamically. The current set of permissions are as described in the table below.
+
+XXXXXX GG: Please review/tweak this and then I'll create the RST table to put here: https://docs.google.com/spreadsheets/d/1z3yUIQs2JaxmUjD5Q8MZSijvbE2o8CoHdvV5Vfj-y4Y/edit#gid=0 
+
+Applicable database fields and tables.
+
+``User`` ``TeamMember`` ``ChannelMember`` Tables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Roles are applied to objects that represents that user’s membership in a context. These are represented by the following objects in the Mattermost data model:
+
+- ChannelMember for a channel context.
+- TeamMember for a team context.
+- User for the system context. 
+
+XXXXXX GG: can we expand on this, what fields are relevant in these tables?
+
```

Comment:
Roles are referenced in the following tables: `User` `TeamMember` `ChannelMember` `Scheme`. For the two `member` tables, it's the `Roles` field that contains *custom* roles and the `SchemeAdmin` and `SchemeUser` booleans that indicate the member object should inherit the respective roles from the relevant scheme (either default or custom one assigned to the relevant team).